support/BRFileService.c

```diff
@@ -260,213 +599,181 @@ fileServiceLoad (BRFileService fs,
+        if (updateVersion &&
+            (version != entityType->currentVersion ||
+             headerVersion != currentHeaderFormatVersion))
+            _fileServiceSave (fs, type, entity, 0);
     }
+    sqlite3_reset (fs->sdbSelectAllStmt);
+    pthread_mutex_unlock (&fs->lock);
 
-    if (0 != fclose (file)) {  fileServiceFailedUnix (fs, bytes, NULL, errno); return; }
+    if (dataBytes != dataBytesBuffer) free (dataBytes);
 
-    free (bytes);
+    return 1;
 }
 
 /// MARK: - Remove, Clear
 
-extern void
+extern int
 fileServiceRemove (BRFileService fs,
                    const char *type,
                    UInt256 identifier) {
     BRFileServiceEntityType *entityType = fileServiceLookupType (fs, type);
-    if (NULL == entityType) { fileServiceFailedImpl (fs, NULL, NULL, "missed type"); return; };
+    if (NULL == entityType)
+        return fileServiceFailedImpl (fs, 0, NULL, NULL, "missed type");
+
+    // Hex-Encode identifier
+    const char *hash = u256hex(identifier);
+
+    sqlite3_status_code status;
+
+    pthread_mutex_lock (&fs->lock);
+
+    status = sqlite3_bind_text (fs->sdbDeleteStmt, 1, type, -1, SQLITE_STATIC);
+    if (SQLITE_OK != status)
+        return fileServiceFailedSDB (fs, 1, status);
+
+    status = sqlite3_bind_text (fs->sdbDeleteStmt, 2, hash, -1, SQLITE_STATIC);
+    if (SQLITE_OK != status)
+        return fileServiceFailedSDB (fs, 1, status);
+
+    if (SQLITE_DONE != sqlite3_step (fs->sdbDeleteStmt))
+        return fileServiceFailedSDB (fs, 1, status);
 
-    char filename[strlen(fs->pathToType) + 1 + strlen(type) + 1 + 2*sizeof(UInt256) + 1];
-    sprintf (filename, "%s/%s/%s", fs->pathToType, type, u256hex(identifier));
+    sqlite3_reset (fs->sdbDeleteStmt);
+    pthread_mutex_unlock (&fs->lock);
 
-    // If failed, then what?
-    remove (filename);
+    return 1;
 }
 
-static void
+static int
 fileServiceClearForType (BRFileService fs,
                          BRFileServiceEntityType *entityType) {
-    DIR *dir;
-    struct dirent *dirEntry;
+    const char *type = entityType->type;
 
-    char dirPath[strlen(fs->pathToType) + 1 + strlen(entityType->type) + 1];
-    sprintf (dirPath, "%s/%s", fs->pathToType, entityType->type);
+    sqlite3_status_code status;
 
-    if (-1 == directoryMake(dirPath) || NULL == (dir = opendir(dirPath))) {
-        fileServiceFailedUnix (fs, NULL, NULL, errno);
-        return;
-    }
+    pthread_mutex_lock (&fs->lock);
 
-    while (NULL != (dirEntry = readdir(dir)))
-        if (dirEntry->d_type == DT_REG)
-            remove (dirEntry->d_name); // If failed, then what?
+    status = sqlite3_bind_text (fs->sdbDeleteAllTypeStmt, 1, type, -1, SQLITE_STATIC);
+    if (SQLITE_OK != status)
+        return fileServiceFailedSDB (fs, 1, status);
 
-    closedir(dir);
+    if (SQLITE_DONE != sqlite3_step (fs->sdbDeleteAllTypeStmt))
+        return fileServiceFailedSDB (fs, 1, status);
+
+    sqlite3_reset (fs->sdbDeleteAllTypeStmt);
```

Comment:
Same comment re: is this sqlite3_reset needed in the early return cases?

Reply:
As above.